### PR TITLE
Added clarity and re-ordered sections

### PR DESCRIPTION
**Pull request by terichadbourne**
- Added 'Key concepts' section to explain Offline First design pattern
- Moved 'Tutorial' and 'Live Demo' above fold
- Clarified what the 'Steps' accomplish
- Reworded intro paragraph
- Commented out 'Learn more' section since it has so far been irrelevant
- Made other updates as made and approved previously on Vanilla implementation

### CODE REVIEW

**Comment by terichadbourne** [comment]
@bradley-holt @lornajane This PR should close issue 107, taking all of the explanatory changes we made yesterday to the Vanilla implementation and applying them to the template for an easier starting point moving forward. 

I tried to mark where the text might change in the cases where an implementation is NOT a PWA, but I might not have the technical knowledge to have gotten that bit correct. 

Also noted where a section should be removed if the resource doesn't exist (live demo, tutorial).

File: templates/README.md [hunk 1]
```diff
@@ -10,9 +10,9 @@ To use this template, view as plain text (it contains comments), then edit as ap
 <!-- fill in the blanks -->
 # Create an Offline First Shopping List with [technologies]
 
-In this code pattern, we will create an Offline First shopping list. Shopping List is an Offline First demo Progressive Web App built using [technologies]. [This app is part of a series of Offline First demo apps, each built using a different stack.](https://github.com/ibm-watson-data-lab/shopping-list).
+This code pattern is a reference implementation of an Offline First shopping list app, built as a Progressive Web App using [technology](technologyURL). [This app is part of a series of Offline First demo apps, each built using a different stack.](https://github.com/ibm-watson-data-lab/shopping-list).
```
Comment (bradley-holt) [inline templates/README.md]: Extra period at the end of the sentence here. That was here from before, but we might as well fix it now.

Comment (bradley-holt) [inline templates/README.md]: Maybe add a comment before this paragraph:

**Comment by lornajane** [comment]
I am +1 especially with the changes @bradley-holt suggested

**Pull request merged**